test/hotspot/gtest/runtime/test_os.cpp

```diff
@@ -924,3 +924,24 @@ TEST_VM(os, open_O_CLOEXEC) {
   ::close(fd);
 #endif
 }
+
+TEST_VM(os, reserve_at_wish_address_shall_not_replace_mappings_smallpages) {
+  char* p1 = os::reserve_memory(M, false, mtTest);
+  ASSERT_NE(p1, (char*)nullptr);
```

Comment:
Do you need casts on nullptr?

Reply:
You are right, it does not. That is an advantage to NULL then. This means I can remove the ASSERT_NULL I introduced a while back.